src/Utils/Startup/windows/WindowsScriptManager.ts

```diff
@@ -0,0 +1,63 @@
+import { ScriptManager } from '../SсriptManager';
+import { StartupOptions } from '../StartupOptions';
+import { PidStore } from '../PidStore';
+import { RegistryHelper } from './RegistryHelper';
+import WinReg from 'winreg';
+import { ChildProcess, spawn } from 'child_process';
+
+export class WindowsScriptManager implements ScriptManager {
+  private readonly _pidStore;
+
+  constructor(pidStore: PidStore) {
+    this._pidStore = pidStore;
+  }
+
+  public getPidIfExists(): Promise<number | undefined> {
+    return this._pidStore.get() || undefined;
+  }
+
+  public install(opts: StartupOptions): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+      const autoRunKey = RegistryHelper.getRunKey();
+      autoRunKey.set(
+        opts.name,
+        WinReg.REG_SZ,
+        this.getRunString(opts),
```

Comment:
To make it start as expected when restarting Windows, we have to set `--daemon` option